interface/forms/vitals/report.php

```diff
@@ -26,125 +27,440 @@ function US_weight($pounds, $mode = 1)
     }
 }
 
+function displayLengthString($value)
+{
+    $lengthInMetric = round(number_format($value * 2.54, 2), 1);
+    return $value . ' in' . ' \ ' . $lengthInMetric . ' cm';
+
+}
+
 function vitals_report($pid, $encounter, $cols, $id, $print = true)
 {
     $count = 0;
+    $display = '';
+    $report = '';
     $data = formFetch("form_vitals", $id);
     $patient_data = getPatientData($GLOBALS['pid']);
     $patient_age = getPatientAge($patient_data['DOB']);
 
     $vitals = "";
-    if ($data) {
-        $vitals .= "<table><tr>";
-
-        foreach ($data as $key => $value) {
```

Comment:
So the vitals form is built as a loop on each column value which future proofs this to allow more field values to be added.  Look at the else clause at the end of the loop.  In this PR the code walks through the form procedurally and does not deal with additional form field values.  That should be addressed.

Reply:
So one of the issues included getting rid of the "table" since that practice for displaying information is outdated while also grouping different measurements together.  I can change this to allow new fields, but it might be difficult to place the fields where they need to be.  For example, the measurements "Weight, Height, BMI, Head Circumference" should be grouped together.  As should Vitals (BP, Respiration, pulse).  I can figure out a clever way of improving this, but with all the new changes this might take some time.  

I'll put some thought into this and come up with a strategy to handle this.  

There are still more improvements to be made, such as getting the dateset from Harriet Lane that gives percentiles for things like BP based on age, height, weight.  That is for later.

There are other datasets that are out there that can greatly improve this, but this is just a start.

I'm open for suggestions while I fix the translations.

Reply:
Are all of the groupings going to be the same across practices?  I think if you group the related data together and then any extra fields / future fields can be just placed in their own group at the end would resolve this problem, right?

Reply:
I believe so.  For example, height and weight aren't really vitals.  BP and pulse are.  I'll make new measurements into their own group that are places above the notes.

Reply:
OK.  I"m looking through the views and such and I can't seem to find where we add new measurements.  It looks like we would have to add code so there really isn't a way to add a new measurement without adding more code.  The views also used the outdated Tables which should probably be replaced with CSS grid.